### PR TITLE
login: bug fixes towards the new browser based login flow

### PR DESCRIPTION
**Pull request by yugangw-msft**
This PR will address:
1. launch right url for ADFS log-in
2. use newer ADAL for correct token cache management
3. handle /favicon.ico request from legacy IE

Mark as [do not merge] as this PR includes a private version of ADAL. I am submitting it to let CI verify no regressions 

- [na] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [na] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yugangw-msft** [comment]
This PR can only get in after a new ADAL release with https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/161

**Comment by rayluo** [comment]
@yugangw-msft ADAL Python 1.0.2 has just been released in both Github and PyPI.

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -1017,21 +1026,31 @@ def _get_authorization_code_worker(tenant, aad_endpoint, resource, results):
         return
 
     # launch browser:
-    url = ('{0}/{1}/oauth2/authorize?response_type=code&client_id={2}'
-           '&redirect_uri={3}&state={4}&resource={5}&prompt=select_account')
-    url = url.format(aad_endpoint, tenant or _COMMON_TENANT, _CLIENT_ID, reply_url, 'code', resource)
+    url = ('{0}/oauth2/authorize?response_type=code&client_id={1}'
+           '&redirect_uri={2}&state={3}&resource={4}&prompt=select_account')
+    url = url.format(authority_url, _CLIENT_ID, reply_url, 'code', resource)
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/_profile.py]: Is there an API in `adal` to construct this URL?

Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/_profile.py]: no

**Pull request merged**